pkg/controller/secretannotator/azure/validate.go

```diff
@@ -0,0 +1,43 @@
+package azure
+
+import (
+	"fmt"
+
+	"github.com/Azure/go-autorest/autorest/adal"
+	"github.com/Azure/go-autorest/autorest/azure"
+)
+
+//go:generate mockgen -source=./validate.go -destination=./mock/validate_generated.go -package=mock
+
+// AdalClient is interface for github.com/Azure/go-autorest/autorest/adal for easier testing
+type AdalClient interface {
```

Comment:
This is needed for mocking for testing. But I think this pkg might not be the best place. And name too. Suggestions are welcome.

Reply:
I think this is okay. This is very azure-specific, and it's already in an azure package. I can't think of an alternative.

Maybe the filename of validate.go could be something like adalclient.go. WDYT?

Reply:
`AdalService`, as its not a client, It set of libraries and methods for AD